nomad/encrypter.go

```diff
@@ -32,6 +32,12 @@ import (
 
 const nomadKeystoreExtension = ".nks.json"
 
+type claimSigner interface {
+	SignClaims(*structs.IdentityClaims) (string, string, error)
+}
+
+var _ claimSigner = &Encrypter{}
```

Comment:
I think you can move this into the `nomad/encrypter_test.go` and it'll have the same effect of asserting the interface, without allocating an `Encrypter` struct. That's how we do the various alloc/task runner hook interface tests.

Reply:
I wanted to firm up my understanding of this pattern in general, so after reading a bunch of the [not-as-clear-as-I'd-like Go spec doc](https://go.dev/ref/spec#Blank_identifier), I asked in our internal Go slack channel. The prevailing view seems to be that this gets thrown away by the compiler (after checking the type), whether in a `_test.go` or not, so I think I'll keep it here as connective tissue right next to the interface and implementation.

Reply:
Sounds good.